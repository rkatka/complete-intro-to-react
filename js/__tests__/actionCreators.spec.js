// @flow
import moxios from 'moxios';
import { setSearchTerm, addAPIData } from '../actionCreators';
import getAPIDetails from '../asyncActions';

const atlanta = {
  title: 'Atlanta',
  year: '2008â€“2013',
  description: 'Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; "Earnest \'Earn\' Marks," an ambitious college drop-out and his estranged cousin, who suddenly becomes a star.',
  poster: 'a.jpg',
  imdbID: 'tt4288182',
  trailer: 'MpEdJ-mmTlY',
  rating: '8.8'
};

test('setSearchTerm', () => {
  expect(setSearchTerm('New York')).toMatchSnapshot();
});

test('addApiData', () => {
  expect(atlanta).toMatchSnapshot();
});

test('getAPIDetails', (done: Function) => {
  moxios.withMock(() => {
    const dispatchMock = jest.fn();
    // invoke the thunk that's returned
    getAPIDetails(atlanta.imdbID)(dispatchMock);
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request
        .respondWith({
          status: 200,
          response: atlanta
        })
        .then(() => {
          expect(request.url).toEqual(`http://localhost:3000/${atlanta.imdbID}`);
          expect(dispatchMock).toBeCalledWith(addAPIData(atlanta));
          done();
        });
    });
  });
});
