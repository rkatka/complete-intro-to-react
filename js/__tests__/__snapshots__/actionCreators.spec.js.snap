// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addApiData 1`] = `
Object {
  "description": "Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; \\"Earnest 'Earn' Marks,\\" an ambitious college drop-out and his estranged cousin, who suddenly becomes a star.",
  "imdbID": "tt4288182",
  "poster": "a.jpg",
  "rating": "8.8",
  "title": "Atlanta",
  "trailer": "MpEdJ-mmTlY",
  "year": "2008â€“2013",
}
`;

exports[`setSearchTerm 1`] = `
Object {
  "payload": "New York",
  "type": "SET_SEARCH_TERM",
}
`;
